tests/bam_io/test_bam_index.h

```diff
@@ -84,15 +84,30 @@ SEQAN_DEFINE_TEST(test_bam_io_bam_index_open)
     BamHeader header;
     readHeader(header, bamFile);
 
-    bool found = true;
-    SEQAN_ASSERT(jumpToRegion(bamFile, found, 0, 1, 10, baiIndex));
-    SEQAN_ASSERT(found);
-    SEQAN_ASSERT(jumpToRegion(bamFile, found, 0, 2, 100, baiIndex));
-    SEQAN_ASSERT(found);
-    SEQAN_ASSERT_NOT(jumpToRegion(bamFile, found, 1, 1, 10, baiIndex));
-    SEQAN_ASSERT_NOT(found);
-    SEQAN_ASSERT(jumpToRegion(bamFile, found, 0, 20, 100, baiIndex));
-    SEQAN_ASSERT_NOT(found);
+    bool hasAlignments = true;
+    BamAlignmentRecord record;
+
+    SEQAN_ASSERT(jumpToRegion(bamFile, hasAlignments, 0, 1, 10, baiIndex));
+    SEQAN_ASSERT_MSG(hasAlignments, "There should be alignments in region [1,10].");
+    readRecord(record, bamFile);
+    SEQAN_ASSERT_MSG(record.beginPos == 0, "Jumping to region [1,10] should go to the first read.");
+
+    SEQAN_ASSERT(jumpToRegion(bamFile, hasAlignments, 0, 2, 100, baiIndex));
+    SEQAN_ASSERT_MSG(hasAlignments, "There should be alignments in region [2,100].");
+    readRecord(record, bamFile);
+    SEQAN_ASSERT_MSG(record.beginPos == 1, "Jumping to region [2,10] should go to the second read.");
+
+    SEQAN_ASSERT(jumpToRegion(bamFile, hasAlignments, 0, 3, 100, baiIndex));
+    SEQAN_ASSERT_MSG(hasAlignments, "There should be alignments in region [3,100].");
+    readRecord(record, bamFile);
+    SEQAN_ASSERT_MSG(record.beginPos == 2, "Jumping to region [3,10] should go to the third read.");
+    SEQAN_ASSERT(atEnd(bamFile));
+
+    SEQAN_ASSERT_NOT(jumpToRegion(bamFile, hasAlignments, 1, 1, 10, baiIndex));
+    SEQAN_ASSERT_NOT_MSG(hasAlignments, "No alignments should be on contig 1.");
+
+    SEQAN_ASSERT(jumpToRegion(bamFile, hasAlignments, 0, 4, 100, baiIndex));
```

Comment:
The tests always have a read starting exactly at the beginning of the region. Maybe we add one more read at pos... 42? and jump to region `[4,100]` expecting success.

Reply:
Good point. There are some other tests that depend on the `small.bam` file that I don't want to break though.. I think I will rewrite the `jumpToRegion` tests using `ex1.bam` instead so I can more easily cover other cases.